app/assets/javascripts/QDMPatient.js

```diff
@@ -103,15 +103,7 @@ QDMPatientSchema.methods.getByProfile = function getByProfile(profile, isNegated
   // If isNegated == true, only return data elements with a negationRationale that is not null.
   // If isNegated == false, only return data elements with a null negationRationale.
   // If isNegated == null, return all matching data elements by type, regardless of negationRationale.
-  const results = this.dataElements.filter(element => (element._type === `QDM::${profile}` || element._type === profile) && (isNegated === null || !!element.negationRationale === isNegated));
-  return results.map((result) => {
-    const removedMongooseItems = new AllDataElements[profile](result).toObject({ virtuals: true });
-    // toObject() will remove all mongoose functions but also remove the schema methods, so we add them back
-    Object.entries(result.schema.methods).forEach(([method_name, method]) => {
-      removedMongooseItems[method_name] = method;
-    });
-    return removedMongooseItems;
-  });
+  return this.dataElements.filter(element => (element._type === `QDM::${profile}` || element._type === profile) && (isNegated === null || !!element.negationRationale === isNegated));
```

Comment:
I don't know why we need to convert Mongoose objects to objects. and then add back methods to the converted object. toObject removed methods from nexted objects as well. if we have to add back methods from nested objects we have to iterate over all the props of the data element and add all the missing methods. I do not think we need that.

Reply:
I can't say that I'm familiar with the history for that approach.  I ran a regression in Cypress and everything works as expected with this new approach.

Reply:
thanks for verifying it!